SimPEG/potential_fields/magnetics/simulation.py

```diff
@@ -281,14 +291,14 @@ def evaluate_integral(self, receiver_location, components):
         R4 = dy1dy1 + dx1dx1
 
         # radius to each cell node
-        r1 = np.sqrt(dz2dz2 + R2) + eps
-        r2 = np.sqrt(dz2dz2 + R1) + eps
-        r3 = np.sqrt(dz1dz1 + R1) + eps
-        r4 = np.sqrt(dz1dz1 + R2) + eps
-        r5 = np.sqrt(dz2dz2 + R3) + eps
-        r6 = np.sqrt(dz2dz2 + R4) + eps
-        r7 = np.sqrt(dz1dz1 + R4) + eps
-        r8 = np.sqrt(dz1dz1 + R3) + eps
+        r1 = np.sqrt(dz2dz2 + R2)
+        r2 = np.sqrt(dz2dz2 + R1)
+        r3 = np.sqrt(dz1dz1 + R1)
+        r4 = np.sqrt(dz1dz1 + R2)
+        r5 = np.sqrt(dz2dz2 + R3)
+        r6 = np.sqrt(dz2dz2 + R4)
+        r7 = np.sqrt(dz1dz1 + R4)
+        r8 = np.sqrt(dz1dz1 + R3)
 
         # compactify argument calculations
         arg1_ = dx1 + dy2 + r1
```

Comment:
Can you try to put the eps on the args instead? Looks like we were not fully protected before...